File: src/main/java/com/vicen/films/domain/Identifiable.java
package com.vicen.films.domain;

import java.util.UUID;

public interface Identifiable {

    UUID getId();

    static <T> UUID getIdFrom(T object) {
        if (object instanceof Identifiable) {
            return ((Identifiable) object).getId();
        } else {
            throw new NotIdentifiableException();
        }
    }

    class NotIdentifiableException extends RuntimeException {
        private NotIdentifiableException() {
            super("Not identifiable object.");
        }
    }

}

File: src/main/kotlin/com/vicen/films/domain/Identifiable.kt
package com.vicen.films.domain

import java.util.*

interface Identifiable {

    val id: UUID

    companion object {
        fun <T> getIdFrom(obj: T) : UUID {
            when (obj) {
                is Identifiable -> return obj.id;
            }
            throw NonIdentifiableException();
        }
    }

    class NonIdentifiableException : RuntimeException("Not identifiable object.")
}

